src/Content/Conversation/Factory/Channel.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+ namespace Friendica\Content\Conversation\Factory;
+
+use Friendica\Model\User;
+use Friendica\Content\Conversation\Entity\Channel as ChannelEntity;
+use Friendica\Core\L10n;
+use Friendica\Database\Database;
+use Psr\Log\LoggerInterface;
+
+final class Channel extends \Friendica\BaseModel
+{
+	/** @var L10n */
+	protected $l10n;
+
+	public function __construct(L10n $l10n, Database $database, LoggerInterface $logger, array $data = [])
+	{
+		parent::__construct($database, $logger, $data);
+
+		$this->l10n = $l10n;
+	}
+	
+	/**
+	 * List of available channels
+	 *
+	 * @param integer $uid
+	 * @return array
+	 */
+	public function getForUser(int $uid): array
+	{
+		$language  = User::getLanguageCode($uid);
+		$languages = $this->l10n->getAvailableLanguages(true);
+
+		$tabs = [
```

Comment:
Now you have an array of a specific kind of entities, it's the perfect time to create a Collection class that will ensure this!

You can copy the file from `src/Collection/Api/Mastodon/Mentions.php` and change the names around for this.

Then it's just a matter of instantiating the collection with the array of entities you created here:
```suggestion
		$tabs = new Channels([
```

Reply:
Please create a Collection for Channel entities.

Reply:
I will. But not today.